### ISSUE TITLE
Setting LiveData within FunSpec

### ISSUE THREAD

**Issue by iainism** [kotest/kotest#1896]
Using Kotest 4.3.1

I'm in the process of attempting to migrate my tests within an Android app to use FunSpec, and I cannot work out how to set a value on MutableLiveData<> when testing my ViewModel **_after_** migration.

With:
```
class ProductDetailsViewModel (private val myRepository: MyRepository) : ViewModel() {
    // Exposed to UI
    val productName = MutableLiveData<String>()

    // ...
}
```

I can successfully test with:
```
@ExtendWith(InstantTaskExecutorExtension::class)
@TestInstance(TestInstance.Lifecycle.PER_METHOD)
class ProductDetailsViewModelTest{
    private val dataSourceMock = mockk<ProductRepository>(relaxed = true)

    private val td = ProductTestData()

    @Test
    fun `DetailsViewModel minimal` () {
        val vm = ProductDetailsViewModel(dataSourceMock)

        vm.productName.value = td.minimalTestProduct.productName
        true shouldBe true  // Nonsense test to get a pass
    }
}
```

I have tried to migrate this to FunSpec as follows:
```
@ExtendWith(InstantTaskExecutorExtension::class)
@TestInstance(TestInstance.Lifecycle.PER_METHOD)
class ProductDetailsViewModelTest : FunSpec({
    val dataSourceMock = mockk<ProductRepository>(relaxed = true)

    val td = ProductTestData()

    val vm = ProductDetailsViewModel(dataSourceMock)

    test("DetailsViewModel minimal") {
        vm.productName.value = td.minimalTestProduct.productName

        true shouldBe true // Nonsense test to get a pass
    }
```

And this gives an exception, which I think is because the InstantTaskExecutorExtension isn't being applied via the FunSepc:
```
android/os/Looper
java.lang.NoClassDefFoundError: android/os/Looper
	at androidx.arch.core.executor.DefaultTaskExecutor.isMainThread(DefaultTaskExecutor.java:77)
	at androidx.arch.core.executor.ArchTaskExecutor.isMainThread(ArchTaskExecutor.java:116)
	at androidx.lifecycle.LiveData.assertMainThread(LiveData.java:461)
	at androidx.lifecycle.LiveData.setValue(LiveData.java:304)
	at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java:50)
```

Is there a way to do what I am trying to do?  Or should I just stick with 'standard' Junit5 tests when trying to work with LiveData?

**Comment by aids61517** [comment]
You should create a ProjectConfig like this

```
object ProjectConfig : AbstractProjectConfig() {
  private val testDispatcher = TestCoroutineDispatcher()

  override fun beforeAll() {
    super.beforeAll()
    setupLiveData()
    //setup whatever you want
  }

  override fun afterAll() {
    super.afterAll()
    resetLiveData()
  }

  private fun setupLiveData() {
    ArchTaskExecutor.getInstance().setDelegate(object : TaskExecutor() {
      override fun executeOnDiskIO(runnable: Runnable) {
        runnable.run()
      }

      override fun postToMainThread(runnable: Runnable) {
        runnable.run()
      }

      override fun isMainThread(): Boolean {
        return true
      }
    })
  }

  private fun resetLiveData() {
    ArchTaskExecutor.getInstance().setDelegate(null)
  }
}
```

**Comment by iainism** [comment]
Thank you so much!

Is there somewhere that this is documented please?  Everything I found when stfw pointed me to using the `@ExtendWith(InstantTaskExecutorExtension::class)` annotation, which clearly doesn't do what is needed in this case.

**Comment by aids61517** [comment]
I'm not sure what InstantTaskExecutorExtension do.

I searched about `@ExtendWith(InstantTaskExecutorExtension::class)` and I found [this](https://stackoverflow.com/questions/48602979/android-livedata-in-androidtest-returns-null) on StackOverflow.

I guess InstantTaskExecutorExtension works based on JUnit 5's lifecycle, and should convert to based on Kotest's lifecycle.

Here is document for Project Config
[https://kotest.io/docs/framework/project-config.html](https://kotest.io/docs/framework/project-config.html)
